combat-trainer.lic

```diff
@@ -65,6 +70,17 @@ class SetupProcess
 
   private
 
+  def blow_warhorn()
```

Comment:
just throwing it out there, how about something like ```  def blow_warhorn()
    return unless @warhorn
    return if Time.now - @last_warhorn < 600
    case bput('tap my warhorn', 'inside your', 'that you are wearing', 'I could not find what you were referring to.')
	when 'inside your'
	  if bput('get my warhorn', 'You get', 'What were you referring to', 'You need a free hand') == 'You get'
        bput('exhale warhorn lure','You sound a series of bursts from the warhorn','Your lungs are tired from having sounded a warhorn so recently.')
        @last_warhorn = Time.now
        waitrt?
        fput("stow #{@warhorn}")
    when 'that you are wearing'
	   if bput('remove my warhorn', 'you list the messages here') == 'successful message'
         bput('exhale warhorn lure','You sound a series of bursts from the warhorn','Your lungs are tired from having sounded a warhorn so recently.')
         @last_warhorn = Time.now
         waitrt?
         fput("stow #{@warhorn}")
	else
	  return
    end
  end```

Reply:
done